jib-build-plan/src/main/java/com/google/cloud/tools/jib/api/buildplan/FileEntriesLayer.java

```diff
@@ -300,6 +315,170 @@ public Builder addEntryRecursive(
       return this;
     }
 
+    /**
+     * Adds an entry to the layer. Only adds the single source file to the exact path in the
+     * container file system.
+     *
+     * <p>For example, {@code newAddEntry(Paths.get("myfile"),
+     * AbsoluteUnixPath.get("/path/in/container"))} adds a file {@code myfile} to the container file
+     * system at {@code /path/in/container}.
+     *
+     * <p>For example, {@code newAddEntry(Paths.get("mydirectory"),
+     * AbsoluteUnixPath.get("/path/in/container"))} adds a directory {@code mydirectory/} to the
+     * container file system at {@code /path/in/container/}. This does <b>not</b> add the contents
+     * of {@code mydirectory}.
+     *
+     * @param sourceFile the source file to add to the layer
+     * @param pathInContainer the path in the container file system corresponding to the {@code
+     *     sourceFile}
+     * @return this
+     */
+    // TODO: Rename this method once deprecated methods have been removed.
+    public Builder newAddEntry(Path sourceFile, AbsoluteUnixPath pathInContainer) {
```

Comment:
In general we try not to do this: Create a new method with a new name and change the name when deprecating an old method. Someone trying to use this new api will inevitably have to change their code again when we rename.

Reply:
Ah that is good to know. Agreed, this would be a hassle for the user of the method.